### ISSUE TITLE
Play sound effect takes ridiculously long to parse/reload

### ISSUE THREAD

**Issue by jaylawl** [SkriptLang/Skript#1091]
Using Skript dev31c

Reproduction:

```
command /test:
    trigger:
    
        play sound "entity.chicken.egg" with volume 1 and pitch 1 at command sender's location for command sender
```
Create this script and add it to your server. Then reload the script and see how long it takes to achieve that. I'm using skUtilities "system milliseconds" expression to output reloading times of scripts. Depending on what kind of expression i use in the play sound effect, reloading it takes between 0.5 and 2 seconds - whereas pretty much all other short scripts like this take < 0 milliseconds to reload.

https://snag.gy/3n0L7A.jpg

Update: i just upgraded to dev32d and the reloading now takes between 0.5-0.6 seconds, opposed to the prior up to 2 seconds.

**Comment by TheBentoBox** [comment]
What do you mean by "Depending on what kind of expression i use in the play sound effect"? Are you saying different syntaxes take longer to reload? If so you may not be using vanilla Skript syntaxes in those cases, as multiple addons have play sound effects that have a very similar syntax to ours.

**Comment by jaylawl** [comment]
@TheBentoBox sorry for being too vague on that. I meant depending on what kinda of player expressions i used. (command sender/{_player}/event-player/etc.). I was also making different tests with functions and other event types, not only just a /test command.

**Comment by Pikachu920** [comment]
to add more info to this, any effect that starts with `play` takes a stupidly long amount of time to parse

**Comment by Pikachu920** [comment]
@TheBentoBox why remove bug? the fact that anything with play takes so long to reload points to some funny business

**Comment by TheBentoBox** [comment]
Because it could just be poor parsing optimization rather than an actual bug. It also could be a bug or some sort of "funny business" as you put it but I shouldn't mark it as such until we know; for now, it's just more correct to label this as an enhancement to the parser, since improving parse speeds is an enhancement, not a bug fix.

**Comment by Pikachu920** [comment]
maybe it would be a good idea to change the syntax to (play|send) as a workaround for this

**Comment by TheBentoBox** [comment]
I agree, it would be a much easier fix that working on the parser.

**Comment by Pikachu920** [comment]
So I looked into this further and I can't reproduce on a server with just Skript. Do you have skrayfall installed by any chance?

**Comment by bensku** [comment]
Not "reproducing" parsing performance issues is a bit tricky. Could be that you have a better computer, so difference is not noticeable.

**Comment by Pikachu920** [comment]
True, however I have definitely experienced this before on my PC, as I noted above.

**Comment by jaylawl** [comment]
@Pikachu920 in case you meant me: yes, i do have SkRayFall installed

**Comment by Misio12320** [comment]
Happens in 1.15.2, and Skript 2.5-alpha1 the long reload.

**Comment by Misio12320** [comment]
@ShaneBeee very please fix this in 2.5-alpha3. Kthxbye :3

**Comment by ShaneBeee** [comment]
> @ShaneBeee very please fix this in 2.5-alpha3. Kthxbye :3

Please don't tag me on issues Im not involved in. Im not the only person on the team. Tagging me means I get an email, even though Im not part of this thread.

**Comment by TheLimeGlass** [comment]
Closing this issue and making the duplicate mentioned above open as it has more recent details. This syntax provided does not cause this issue anymore and reloads in 17ms (That's very good)
